crates/openvino/src/prepostprocess.rs

```diff
@@ -0,0 +1,318 @@
+    pub fn new(model: &Model) -> Result<Self> {
+        let mut preprocess = std::ptr::null_mut();
+        try_unsafe!(ov_preprocess_prepostprocessor_create(
+            model.instance,
+            std::ptr::addr_of_mut!(preprocess)
+        ))?;
+
+        Ok(Self {
+            instance: preprocess,
+        })
+    }
+
+    /// Retrieves the input information by index.
+    pub fn get_input_info_by_index(&self, index: usize) -> Result<PreprocessInputInfo> {
+        let mut input_info = std::ptr::null_mut();
+        try_unsafe!(ov_preprocess_prepostprocessor_get_input_info_by_index(
+            self.instance,
+            index,
+            std::ptr::addr_of_mut!(input_info)
+        ))?;
+
+        Ok(PreprocessInputInfo {
+            instance: input_info,
+        })
+    }
+
+    /// Retrieves the input information by name.
+    pub fn get_input_info_by_name(&self, name: &str) -> Result<PreprocessInputInfo> {
+        let mut input_info = std::ptr::null_mut();
+        let c_layout_desc = CString::new(name).unwrap();
+        try_unsafe!(ov_preprocess_prepostprocessor_get_input_info_by_name(
+            self.instance,
+            c_layout_desc.as_ptr(),
+            std::ptr::addr_of_mut!(input_info)
+        ))?;
+
+        Ok(PreprocessInputInfo {
+            instance: input_info,
+        })
+    }
+
+    /// Retrieves the output information by name.
+    pub fn get_output_info_by_name(&self, name: &str) -> Result<PreprocessOutputInfo> {
+        let mut output_info = std::ptr::null_mut();
+        let c_layout_desc = CString::new(name).unwrap();
+        try_unsafe!(ov_preprocess_prepostprocessor_get_output_info_by_name(
+            self.instance,
+            c_layout_desc.as_ptr(),
+            std::ptr::addr_of_mut!(output_info)
+        ))?;
+
+        Ok(PreprocessOutputInfo {
+            instance: output_info,
+        })
+    }
+
+    /// Retrieves the output information by index.
+    pub fn get_output_info_by_index(&self, index: usize) -> Result<PreprocessOutputInfo> {
+        let mut output_info = std::ptr::null_mut();
+        try_unsafe!(ov_preprocess_prepostprocessor_get_output_info_by_index(
+            self.instance,
+            index,
+            std::ptr::addr_of_mut!(output_info)
+        ))?;
+
+        Ok(PreprocessOutputInfo {
+            instance: output_info,
+        })
+    }
+
+    /// Retrieves the input information.
+    pub fn get_input_info(&self) -> Result<PreprocessInputInfo> {
+        let mut input_info = std::ptr::null_mut();
+        try_unsafe!(ov_preprocess_prepostprocessor_get_input_info(
+            self.instance,
+            std::ptr::addr_of_mut!(input_info)
+        ))?;
+        assert!(!input_info.is_null());
+
+        Ok(PreprocessInputInfo {
+            instance: input_info,
+        })
+    }
+
+    /// Builds model with all steps from pre/postprocessing.
+    pub fn build(&self, new_model: &mut Model) -> Result<()> {
```

Comment:
From my understanding, it doesn't need owned `Model`, it simply builds a new model based on all the preprocessor steps.
`ov_preprocess_prepostprocessor_build(const ov_preprocess_prepostprocessor_t *preprocess, ov_model_t **model)`
I've changed the `build` function and the tests which use it to make it so that it now returns a `new_model`
Here's[ C version](https://github.com/openvinotoolkit/openvino/blob/a7c7de4c65cdb683451d59bf207e93c444030d0d/samples/c/hello_classification/main.c#L202) of `hello_classification` where it does `preprocessor_build`